### ISSUE TITLE
`yii\web\Request` does not pass full `content-type` to `RequestParserInterface::parse()`

### ISSUE THREAD

**Issue by klimov-paul** [yiisoft/yii2#12463]
At `Request::getBodyParams()` 'content-type' valu can be striped from the sensitive information.
See:
https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L389-L392

here additional info of the 'content-type' is stripped in order to be able to find correct parser.

However, while invoking `RequestParserInterface::parse()` same limited content-type is passed:
https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L399

Content-type may contain some sensitive information. For example: in case `multipart/form-data` is passed in contains the `boundary`, which is necessary for correct body parsing:

```
Content-Type: multipart/form-data; boundary=---------------------------22472926011618
```

I suggest the FULL content-type value should be passed down to the `RequestParserInterface::parse()`.

**Comment by klimov-paul** [comment]
Fix may be considered as BC break as some external parser relies on current policy around 'content-type'

**Comment by klimov-paul** [comment]
Resolved by commit ed0603b706635cd86ae70e6ea256aa6fa000166f
